### ISSUE TITLE
DirectX 8?

### ISSUE THREAD

**Issue by MajorLunaC** [Winetricks/winetricks#546]
I know it's a bit old, but I've found some older programs that require DirectX 8 specifically. I've heard that DirectX 9 is somewhat backwards compatible with DirectX 8, but that doesn't help it run.

I tried to install a version or two of DirectX 8 manually with no real success. Can some support for DirectX 8 be added? Or is there another solution around this?

**Comment by austin987** [comment]
On Sep 21, 2015 12:01 PM, "MajorLunaC" <notifications@github.com> wrote:
>
> I know it's a bit old, but I've found some older programs that require
DirectX 8 specifically. I've heard that DirectX 9 is somewhat backwards
compatible with DirectX 8, but that doesn't help it run.
>
> I tried to install a version or two of DirectX 8 manually with no real
success. Can some support for DirectX 8 be added? Or is there another
solution around this?
>
> —
> Reply to this email directly or view it on GitHub.

Native DirectX itself isn't usable, for dx9 we're using the redistributable
portions. I'm not aware of anything similar for dx8. Native d3d8.dll isn't
usable in Wine. I don't know of any portion of d3d8 that would be usable.

**Comment by MajorLunaC** [comment]
That's unfortunate. I'll just have to hope somebody will one day be interested enough to develop Wine's d3d8.dll, although with window$ versions being spat out so fast and new DirectX versions on the way, that wont happen anytime soon I'm guessing.

Although with the backwards compatibility, I wonder if some of dx9 can just be copied over to dx8. Easier said than done, I'm sure.

**Comment by austin987** [comment]
On Sep 21, 2015 6:32 PM, "MajorLunaC" <notifications@github.com> wrote:
>
> That's unfortunate. I'll just have to hope somebody will one day be
interested enough to develop Wine's d3d8.dll, although with window$
versions being spat out so fast and new DirectX versions on the way, that
wont happen anytime soon I'm guessing.
>
> Although with the backwards compatibility, I wonder if some of dx9 can
just be copied over to dx8. Easier said than done, I'm sure.
>
> —
> Reply to this email directly or view it on GitHub.

Do you have an upstream bug I can refer to?

**Comment by MajorLunaC** [comment]
I'm not quite sure what you mean. A bug in Wine or a bug in the program I'm running?

When running the program, I get a "DirectX8 failed to initialize. Exiting." pop-up and it exits with no terminal errors.

All I get in the terminal when enabling d3d8.dll builtin is some fixmes that weren't there before:
http://pastebin.com/RbSXDQ8w

As for Wine itself, all I can find is a rare few reference pages. Nothing about any recent bugs:
http://wine-wiki.org/index.php/Wine_DirectX_status
http://wiki.winehq.org/DirectX
https://forum.winehq.org/viewtopic.php?t=572

**Comment by austin987** [comment]
I was asking if you've filed an upstream Wine bug for whatever game you're trying to run. Native d3d8.dll isn't usable, and I don't know of anything else related to DX8 that winetricks could provide.

**Comment by qwertychouskie** [comment]
Since Wine does not support DX8, can this be closed?

**Comment by austin987** [comment]
Wine definitely has support for directx8

**Comment by MajorLunaC** [comment]
Oops! I forgot I posted this. I found the solution after a few months of testing: DX8 is supported by Wine. If a program says it isn't, try a new clean wine prefix with no modifications whatsoever, no wine staging, no winetricks. If that doesn't work, try with wine staging and winetricks. One way or another, Direct X 8 DOES work in normal unmodified Wine, but staging or some winetricks could be getting in the way.
